app/main.py

```diff
@@ -1,3 +1,56 @@
+import operator
+from typing import Union, Callable
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        if not isinstance(km, (int, float)):
+            raise TypeError("km must be a number")
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", float]) -> "Distance":
```

Comment:
You could be more consistent with typing - ints can be used for addition as well.